eureka/lib/masking.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+from scipy.interpolate import NearestNDInterpolator
+
+__all__ = ['interpolating_row', 'data_quality_mask',
+           'interpolating_image']
+
+
+def interpolating_image(data, mask):
+    """
+    Uses `scipy.interpolate.NearestNDInterpolator` to
+    fill in bad pixels/cosmic rays/whichever mask you
+    decide to pass in.
+
+    Parameters
+    ----------
+    data : np.ndarray
+       Image frame.
+    mask : np.ndarray
+       Mask of integers or boolean values, where values
+       greater than 0/True are bad pixels.
+
+    Returns
+    -------
+    cleaned : np.ndarray
+       Array of shape `data` which now has interpolated
+       values over the bad masked pixels.
+    """
+    def interpolate(d, m):
+        try:
+            m = m is False
```

Comment:
m = ~m is an order of magnitude faster for booleans. Also note that this line will end up getting run for integer arrays without error and will give the exact opposite behaviour as expected. And the m = ~m suggestion won't work for integer arrays

Reply:
This isn't my function and I've never run it. I was making changes based on the `flake8` command (since there were a bunch not relevant to my stuff). I've updated it for now, but may want to double check with the original author if this will break things down the line.

Reply:
GitHub shows that this file didn't exist before this PR though - the GitHub history shows that you wrote this 2 months ago with commit e374f053b3b51ee89ba1397a3e9972cf8bf3ea47. Do we no longer need this function?